homeassistant/components/fully_kiosk/entity.py

```diff
@@ -23,6 +23,7 @@ def __init__(self, coordinator: FullyKioskDataUpdateCoordinator) -> None:
             manufacturer=coordinator.data["deviceManufacturer"],
             model=coordinator.data["deviceModel"],
             sw_version=coordinator.data["appVersionName"],
-            configuration_url=f"http://{coordinator.data['ip4']}:2323",
+            configuration_url=f"http{'s' if coordinator.use_ssl else ''}://"
```

Comment:
This feels hacky (and spans multiple lines inline, which we should avoid).

It might be better to build the URL, for example:

- Add import of YARL:
  ```py
  from yarl import URL
  ```

- Build URL:
  ```py
  url = URL.build(
      scheme="https" if coordinator.use_ssl else "http",
      host=coordinator.data["ip4"],
      port=2323,
  )
  ```

- Assign in device info:
  ```py
  configuration_url=str(URL),
  ```

Reply:
Suggestion was implemented.